csrc/multidevice/utils.cpp

```diff
@@ -204,22 +184,49 @@ int64_t getShardedLogicalAxis(
           def);
     }
   }
-  return logical_id_to_axis.at(id);
 }
 
-int64_t getShardedLoopAxis(
+} // namespace
+
+int64_t getShardedLogicalAxis(
     const TensorView* tv,
     const ParallelType parallel_type) {
-  NVF_ERROR(
-      isParallelTypeDeviceDim(parallel_type),
-      "Expect a DID but found: ",
-      parallel_type);
-  for (auto&& [index, loop_id] : enumerate(tv->getLoopDomain())) {
-    if (loop_id->getParallelType() == parallel_type) {
-      return index;
+  IterDomain* parallel_id = getShardedIterDomain(tv, parallel_type);
+  if (parallel_id == nullptr) {
+    return -1;
+  }
+
+  return getDominatingLogicalAxis(tv, parallel_id);
+}
+
+IterDomain* getShardedIterDomain(
```

Comment:
This replaces getShardedLogicalAxis so I can use it in unshardedSizes.

Reply:
Refactoring `getShardedLogicalAxis` and removing `getShardedLoopAxis` can also be a standalone PR which does not need to modify `unshardedSizes`. This will make the remaining changes specific to stream evaluation in HostIR. It also moves renaming in schedulers and tests to a separate PR